cdap-app-fabric/src/test/java/co/cask/cdap/internal/profile/ProfileMetricScheduledServiceTest.java

```diff
@@ -0,0 +1,91 @@
+import co.cask.cdap.common.utils.Tasks;
+import co.cask.cdap.internal.AppFabricTestHelper;
+import co.cask.cdap.proto.id.NamespaceId;
+import co.cask.cdap.proto.id.ProfileId;
+import co.cask.cdap.proto.id.ProgramRunId;
+import com.google.common.collect.ImmutableMap;
+import com.google.inject.Injector;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+public class ProfileMetricScheduledServiceTest {
+  private static Injector injector;
+
+  @BeforeClass
+  public static void setupClass() {
+    injector = AppFabricTestHelper.getInjector();
+  }
+
+  @Test
+  public void testProfileMetrics() throws Exception {
+    ProgramRunId runId = NamespaceId.DEFAULT.app("myApp").workflow("myProgram").run(RunIds.generate());
+    ProfileId profileId = NamespaceId.DEFAULT.profile("myProfile");
+    MetricsCollectionService collectionService = injector.getInstance(MetricsCollectionService.class);
+    MetricStore metricStore = injector.getInstance(MetricStore.class);
+
+    // There are 60 nodes, we emit the metrics each 2 seconds, so each time the node minute should go up by 2 min
+    ProfileMetricScheduledService scheduledService = new ProfileMetricScheduledService(collectionService, runId,
+                                                                                       profileId, 60, 2);
+    scheduledService.startAndWait();
+    // sleep for 6 seconds and expect 3 increment about the metrics
+    TimeUnit.SECONDS.sleep(6);
+    Tasks.waitFor(6L, () -> getMetric(metricStore, runId, profileId,
+                                      "system." + Constants.Metrics.Program.PROGRAM_NODE_MINUTES),
+                  1, TimeUnit.SECONDS);
+    scheduledService.stopAndWait();
+  }
+
+  private long getMetric(MetricStore metricStore, ProgramRunId programRunId, ProfileId profileId, String metricName) {
+    ImmutableMap<String, String> tags = ImmutableMap.<String, String>builder()
+      .put(Constants.Metrics.Tag.PROFILE_SCOPE, profileId.getScope().name())
+      .put(Constants.Metrics.Tag.PROFILE, profileId.getScopedName())
+      .put(Constants.Metrics.Tag.NAMESPACE, programRunId.getNamespace())
+      .put(Constants.Metrics.Tag.PROGRAM_TYPE, programRunId.getType().getPrettyName())
+      .put(Constants.Metrics.Tag.APP, programRunId.getApplication())
+      .put(Constants.Metrics.Tag.PROGRAM, programRunId.getProgram())
+      .build();
+
+    MetricDataQuery query = new MetricDataQuery(0, 0, Integer.MAX_VALUE, metricName, AggregationFunction.SUM,
+                                                tags, new ArrayList<>());
+    Collection<MetricTimeSeries> result = metricStore.query(query);
+    if (result.isEmpty()) {
+      return 0;
+    }
+    List<TimeValue> timeValues = result.iterator().next().getTimeValues();
+    if (timeValues.isEmpty()) {
+      return 0;
+    }
+    return timeValues.get(0).getValue();
+  }
+}
```

Comment:
missing newline

Reply:
These changes lost when I did the rebase, added in the fix build commit